docs/docs/system-administration/architecture.md

```diff
@@ -0,0 +1,11 @@
+---
+title: Architecture
+---
+
+## Self Hosted / On Prem
+
+![Image](/img/self-hosted-architecture.svg)
+
+## SaaS
+
+![Image](/img/saas-architecture.svg)
```

Comment:
Looks like the updated diagram for the SaaS architecture is not part of this PR yet?

Reply:
<img width="1013" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/14089968/dc35a034-0c7d-4210-b050-34ea83bd48c4">